File: src/test/console/java/src/main/java/web/FormAuthenticationPage.java
package web;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.WebDriver;

class FormAuthenticationPage extends Base {
  // Page elements
  @FindBy(id = "username")
  public WebElement usernameText;
  @FindBy(id = "password")
  public WebElement passwordText;
  @FindBy(xpath= "//form[@name=\"login\"]")
  public WebElement loginForm;
  @FindBy(xpath= "//div[@id=\"flash\"]")
  public WebElement message;

  public FormAuthenticationPage(WebDriver driver, long timeoutInSeconds) {
    super(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/login");
  }

  public void open() throws Exception {
    super.open();
    this.waitForElementVisible(usernameText);
  }

  public void login(String username, String password)  throws  Exception {
    this.waitForElementVisible(usernameText);
    usernameText.sendKeys(username);
    passwordText.sendKeys(password);
    loginForm.submit();
  }

  public String getMessage() throws Exception {
    waitForElementVisible(message);
    return message.getText();
  }
}

File: src/test/console/dotnet/Jsonwired/web/FormAuthenticationPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace web {
	class FormAuthenticationPage : Base {
		// Page elements
		[FindsBy(How = How.Id, Using = "username")]
		public IWebElement usernameText ;
		[FindsBy(How = How.Id, Using = "password")]
		public IWebElement passwordText ;
		[FindsBy(How = How.XPath, Using = "//form[@name=\"login\"]")]
		public IWebElement loginForm ;
		[FindsBy(How = How.XPath, Using = "//div[@id=\"flash\"]")]
		public IWebElement message ;

		public FormAuthenticationPage(IWebDriver driver, long timeoutInSeconds):
			base(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/login")	{
			}

		public override void open() {
			base.open();
			this.waitForElementVisible(usernameText);
		}

  	public void login(string username, string password) {
			usernameText.SendKeys(username);
			passwordText.SendKeys(password);
			loginForm.Submit();
		}

  	public string getMessage() {
			this.waitForElementVisible(message);
			return message.Text;
		}
	}
}
